src/txn/txn.c

```diff
@@ -318,6 +318,63 @@ __wt_txn_bump_snapshot(WT_SESSION_IMPL *session)
     __txn_get_snapshot_int(session, false);
 }
 
+/*
+ * __wt_txn_copy_and_bump_snapshot --
+ *     Take backup of the existing snapshot and allocate a new snapshot.
+ */
+void
+__wt_txn_copy_and_bump_snapshot(WT_SESSION_IMPL *session)
+{
+    WT_TXN *txn;
+
+    txn = session->txn;
+
+    /* Initialize the backup snapshot variables. */
+    txn->bkp_snap_min = WT_TXN_MAX;
+    txn->bkp_snap_max = WT_TXN_MAX;
+    txn->bkp_snapshot_count = 0;
+
+    /* Take the backup of the snapshot only if the snapshot count is greater than 0. */
+    if (txn->snapshot_count > 0) {
+        txn->bkp_snap_max = txn->snap_max;
```

Comment:
I think this is not correct. Even the snapshot count is 0, we can have a valid snap min and snap max. It just means there is no parallel transaction running but we should still record the snap min and snap max.

Reply:
Fixed in e677c292f2bdf9f92fc2e5827e8466d65624f41b